### ISSUE TITLE
RFE: TO&E depth doesn't allow full army group organization

### ISSUE THREAD

**Issue by OrbMonky** [MegaMek/mekhq#4763]
![image](https://github.com/user-attachments/assets/eacb35cf-63cb-4373-bfbf-a586829a1060)
If I add anything at the company level, it shows up but if I save and reload it goes away. It appears the 'depth' of the table is limited by one too many for full replication of things like SLDF Army Group structuring. Not a major gripe as you can get creative with condensing, but if it's as easy as just adding extra depth of nesting perhaps it is an easy improvement to what is allowed.

To summarize: If I add any further levels of organization at the 8th level (Company), those appear but are not retained in saves. This creates organizational headaches and seems to be a possibly arbitrary limitation. Ideally, we would be able to scale as we prefer, but a depth of 9 or 10 levels of organization would be preferable for full scale setups in the TO&E.

**Comment by IllianiCBT** [comment]
The reason this doesn’t work is being you have an ‘illegal’ TOE, insofar as mhq is concerned, as there isn’t a formation level above ‘army’ and it counts your origin node as the top level. So if you rearrange your TOE factoring in the origin node it’ll work without issue :)

Otherwise, this is working as intended

**Comment by OrbMonky** [comment]
I'm seeing on Sarna that at least for SLDF, there is an Army Group above Army https://www.sarna.net/wiki/Star_League_Defense_Force

![image](https://github.com/user-attachments/assets/e5145455-5a7b-4791-bd11-78ff1c1b6b99)

I understand Sarna is not necessarily canon, but it seems Army Groups are indeed a thing at least for some militaries at some points within the canon timeline - consisting of several army-scale units sharing broader strategic objectives.

This is a minor nitpick overall, but if it is possible and simple to accomplish, it would be appreciated.

**Comment by HammerGS** [comment]
The SLDF is the gold standard, and while I think it's insane. We should also look closely into how the houses organize. Davion and Kurita have some pretty deep organizations.

**Comment by IllianiCBT** [comment]
Now that we have the base system working as intended, it wouldn't be too difficult to include faction-specific stuff. I set it up so that it was expandable, using Enums and the like

**Comment by OrbMonky** [comment]
One thing to consider for truly taking into account house composition is if a player wants an organization level that is the house itself, if we're considering a top to bottom approach.

**Comment by IllianiCBT** [comment]
The counter argument is that the TOE is meant to represent the organization of military assets. By all accounts ‘House Steiner’ isn’t a part of the House Steiner TO&E.

**Comment by pheonixstorm** [comment]
iirc someone did a test of the entire SLDF in a much older version of HQ. Needless to say it was VERY slow to load and not very stable. Also, it wasn't very practical either.

**Comment by IllianiCBT** [comment]
Closing as resolved
